contracts/src/Functions.sol

```diff
@@ -65,7 +66,20 @@ library Functions {
             revert InvalidAmount();
         }
 
+        uint256 balanceBefore = IERC20(token).balanceOf(agent);
         IERC20(token).safeTransferFrom(sender, agent, amount);
+
+        uint256 balanceAfter = IERC20(token).balanceOf(agent);
+        if (balanceAfter < balanceBefore) {
+            revert InvalidAmount();
+        }
+
+        uint256 delta = balanceAfter - balanceBefore;
+        if (delta == 0 || delta > type(uint128).max) {
+            revert InvalidAmount();
+        }
+
+        transferredAmount = uint128(delta);
```

Comment:
Another option would be to simply revert if the delta does not match the expected amount. Notably, DeFi protocols like Uniswap do not support these edge cases either. https://docs.uniswap.org/concepts/protocol/integration-issues

Reply:
I like this, just revert and not support these kinds of tokens at all. @vgeddes ?

The only bad thing with this route, since we have already accepted PAXG, is that if they enabled fees on transfer and it reverted if we detected that. Then all PAXG would essentially never be allowed to be unlocked.

Reply:
So I have analyzed a large portion of bridged transfers from mainet(10k, ~80%), and we have not run this case. So I think it would be safe to what you suggest and revert @yrong.

Reply:
Agreed, just revert if delta != amount

Reply:
Resolved in [49bdba9](https://github.com/Snowfork/snowbridge/pull/1636/commits/49bdba92f353afeba5b5f5667dad6c25d3645bcf)